janus_core/helpers/struct_io.py

```diff
@@ -0,0 +1,291 @@
+    model_path: Optional[PathLike] = None,
+    calc_kwargs: Optional[dict[str, Any]] = None,
+) -> None:
+    """
+    Configure calculator and attach to structure(s).
+
+    Parameters
+    ----------
+    struct : Optional[MaybeSequence[Atoms]]
+        ASE Atoms structure(s) to attach calculators to.
+    arch : Architectures
+        MLIP architecture to use for calculations. Default is "mace_mp".
+    device : Devices
+        Device to run model on. Default is "cpu".
+    model_path : Optional[PathLike]
+        Path to MLIP model. Default is `None`.
+    calc_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to the selected calculator. Default is {}.
+    """
+    calc_kwargs = calc_kwargs if calc_kwargs else {}
+
+    calculator = choose_calculator(
+        arch=arch,
+        device=device,
+        model_path=model_path,
+        **calc_kwargs,
+    )
+
+    if isinstance(struct, Sequence):
+        for image in struct:
+            image.calc = copy(calculator)
+    else:
+        struct.calc = calculator
+
+
+def input_structs(
+    struct: Optional[MaybeSequence[Atoms]] = None,
+    *,
+    struct_path: Optional[PathLike] = None,
+    read_kwargs: Optional[ASEReadArgs] = None,
+    sequence_allowed: bool = True,
+    arch: Architectures = "mace_mp",
+    device: Devices = "cpu",
+    model_path: Optional[PathLike] = None,
+    calc_kwargs: Optional[dict[str, Any]] = None,
+    set_calc: Optional[bool] = None,
+    logger: Optional[logging.Logger] = None,
+) -> MaybeSequence[Atoms]:
+    """
+    Read input structures and/or attach MLIP calculators.
+
+    Parameters
+    ----------
+    struct : Optional[MaybeSequence[Atoms]]
+        ASE Atoms structure(s) to simulate. Required if `struct_path` is None. Default
+        is None.
+    struct_path : Optional[PathLike]
+        Path of structure to simulate. Required if `struct` is None. Default is None.
+    read_kwargs : ASEReadArgs
+        Keyword arguments to pass to ase.io.read. Default is {}.
+    sequence_allowed : bool
+        Whether a sequence of Atoms objects is allowed. Default is True.
+    arch : Architectures
+        MLIP architecture to use for calculations. Default is "mace_mp".
+    device : Devices
+        Device to run model on. Default is "cpu".
+    model_path : Optional[PathLike]
+        Path to MLIP model. Default is `None`.
+    calc_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to the selected calculator. Default is {}.
+    set_calc : Optional[bool]
+        Whether to set (new) calculators for structures.  Default is True if
+        `struct_path` is specified or `struct` is missing calculators, else default is
+        False.
+    logger : Optional[logging.Logger]
+        Logger if log file has been specified. Default is None.
+
+    Returns
+    -------
+    MaybeSequence[Atoms]
+        Structure(s) with attached MLIP calculators.
+    """
+    read_kwargs = read_kwargs if read_kwargs else {}
```

Comment:
Was there a function for this?

Reply:
We have `none_to_dict`, which iterates over a sequence of dicts, but I'm not sure calling it for one dict would cleaner than this (we'd also need to make it work for non-sequences)